src/ui/public/filter_bar/filter_bar.js

```diff
@@ -46,6 +69,36 @@ define(function (require) {
           }
         };
 
+        var privateFieldRegexp = /(^\$|meta)/;
+        $scope.startEditingFilter = function (source) {
+          var model = _.cloneDeep(source);
+          var filterType;
+
+          //Hide private properties and figure out what type of filter this is
+          _.each(model, function (value, key) {
+            if (key.match(privateFieldRegexp)) {
+              delete model[key];
+            } else {
+              filterType = key;
+            }
+          });
+
+          $scope.editingFilter = {
```

Comment:
I like how you've captured all of the editing metadata in one spot, but why not include the error in this object? (`editingFilter.error` vs `editingFilterError`)